exporter/kafkaexporter/factory_test.go

```diff
@@ -122,15 +122,81 @@ func TestWithMarshalers(t *testing.T) {
 	})
 }
 
-type customMarshaler struct {
+type customTracesMarshaler struct {}
+
+var _ TracesMarshaler = (*customTracesMarshaler)(nil)
+
+func (c customTracesMarshaler) Marshal(_ pdata.Traces, topic string) ([]*sarama.ProducerMessage, error) {
+	panic("implement me")
+}
+
+func (c customTracesMarshaler) Encoding() string {
+	return "custom"
+}
+
+func TestWithLogsMarshalers(t *testing.T) {
+	cm := &customLogsMarshaler{}
+	f := NewFactory(WithLogsMarshalers(cm))
+	cfg := createDefaultConfig().(*Config)
+	// disable contacting broker
+	cfg.Metadata.Full = false
+
+	t.Run("custom_encoding", func(t *testing.T) {
+		cfg.Encoding = cm.Encoding()
+		exporter, err := f.CreateLogsExporter(context.Background(), componenttest.NewNopExporterCreateSettings(), cfg)
+		require.NoError(t, err)
+		require.NotNil(t, exporter)
+	})
+	t.Run("default_encoding", func(t *testing.T) {
+		cfg.Encoding = defaultEncoding
+		exporter, err := f.CreateLogsExporter(context.Background(), componenttest.NewNopExporterCreateSettings(), cfg)
+		require.NoError(t, err)
+		assert.NotNil(t, exporter)
+	})
+}
+
+type customLogsMarshaler struct {}
+
+var _ LogsMarshaler = (*customLogsMarshaler)(nil)
+
+func (c customLogsMarshaler) Marshal(logs pdata.Logs, topic string) ([]*sarama.ProducerMessage, error) {
+	panic("implement me")
```

Comment:
I really worry about panicing in tests, it isn't go idiomatic and not discouraged by the contribution guide. 

Can I ask what you're wanting to achieve here?